gui/elements/qt/_layer_buttons.py

```diff
@@ -0,0 +1,110 @@
+from PyQt5.QtCore import Qt
+from PyQt5.QtGui import QIcon
+from PyQt5.QtWidgets import (QWidget, QVBoxLayout, QHBoxLayout, QPushButton,
+                             QFrame, QCheckBox)
+from os.path import join
+from ...resources import resources_dir
+
+path_delete = join(resources_dir, 'icons', 'delete.png')
+path_add = join(resources_dir, 'icons', 'add.png')
+path_off = join(resources_dir, 'icons', 'annotation_off.png')
+path_on = join(resources_dir, 'icons', 'annotation_on.png')
+
+
+class QtLayerButtons(QFrame):
+    def __init__(self, layers):
+        super().__init__()
+
+        self.layers = layers
+        self.annotationCheckBox = QtAnnotationCheckBox(self.layers)
+        self.deleteButton = QtDeleteButton(self.layers)
+        self.addButton = QtAddButton(self.layers)
+
+        layout = QHBoxLayout()
+        layout.addWidget(self.annotationCheckBox)
+        layout.addStretch(0)
+        layout.addWidget(self.addButton)
+        layout.addWidget(self.deleteButton)
+        self.setLayout(layout)
+
+
+class QtDeleteButton(QPushButton):
+    def __init__(self, layers):
+        super().__init__()
+
+        self.layers = layers
+        self.setIcon(QIcon(path_delete))
+        self.setFixedWidth(28)
+        self.setFixedHeight(28)
+        self.setToolTip('Delete layers')
+        self.setAcceptDrops(True)
+        styleSheet = """QPushButton {background-color:lightGray;
+            border-radius: 3px;}
+            QPushButton:pressed {background-color:rgb(0, 153, 255);
+            border-radius: 3px;}
+            QPushButton:hover {background-color:rgb(0, 153, 255);
+            border-radius: 3px;}"""
+        self.setStyleSheet(styleSheet)
+        self.clicked.connect(self.layers.remove_selected)
+
+    def dragEnterEvent(self, event):
+        event.accept()
+        self.hover = True
+        self.update()
+
+    def dragLeaveEvent(self, event):
+        event.ignore()
+        self.hover = False
+        self.update()
+
+    def dropEvent(self, event):
+        event.setDropAction(Qt.CopyAction)
+        event.accept()
+
+
+class QtAddButton(QPushButton):
+    def __init__(self, layers):
+        super().__init__()
+
+        self.layers = layers
+        self.setIcon(QIcon(path_add))
+        self.setFixedWidth(28)
+        self.setFixedHeight(28)
+        self.setToolTip('Add layer')
+        styleSheet = """QPushButton {background-color:lightGray;
+                        border-radius: 3px;}
+                        QPushButton:pressed {background-color:rgb(0, 153, 255);
+                        border-radius: 3px;}
+                        QPushButton:hover {background-color:rgb(0, 153, 255);
+                        border-radius: 3px;}"""
```

Comment:
Given that this one is identical to the above, I'm thinking this should be a constant.

Reply:
I have pulled it out of the two classes and defined it at the top of the file, let me know if that was what you expected. Should I make it all-caps too?

Reply:
I'm ambivalent on all caps for constants, so do as you prefer. =)

Reply:
i'll leave it